qiskit/primitives/containers/sampler_pub.py

```diff
@@ -113,6 +114,13 @@ def coerce(cls, pub: SamplerPubLike, shots: int | None = None) -> SamplerPub:
         if isinstance(pub, QuantumCircuit):
             return cls(circuit=pub, shots=shots, validate=True)
 
+        if isinstance(pub, CircuitInstruction):
+            raise ValueError(
+                f"An invalid Sampler pub-like was given ({pub}). If you want to run a circuit, "
+                "you need to wrap a circuit with `[]` like `sampler.run([circuit])` "
+                "instead of `sampler.run(circuit)`."
```

Comment:
```suggestion
                f"An invalid Sampler pub-like was given ({pub}). If you want to run a single circuit, "
                "you need to wrap the circuit with `[]` like `sampler.run([circuit])` "
                "instead of `sampler.run(circuit)`."
```